File: src/main/kotlin/customfit/ai/kotlinclient/network/connection/ConnectionInformation.kt
package customfit.ai.kotlinclient.network.connection

/** Provides detailed information about the connection state. */
data class ConnectionInformation(
        /** The current connection status. */
        val status: ConnectionStatus,

        /** When true, indicates the SDK was put in offline mode intentionally. */
        val isOfflineMode: Boolean,

        /** Last connection error message, if any. */
        val lastError: String? = null,

        /** Timestamp of the last successful connection in milliseconds. */
        val lastSuccessfulConnectionTimeMs: Long = 0,

        /** Number of consecutive connection failures. */
        val failureCount: Int = 0,

        /** Time of the next reconnection attempt in milliseconds. */
        val nextReconnectTimeMs: Long = 0
)

File: customfit-flutter-client-sdk/lib/src/network/connection/connection_information.dart
import 'connection_status.dart';

/// Detailed info about the SDK’s connection state
class ConnectionInformation {
  final ConnectionStatus status;
  final bool isOfflineMode;
  final String? lastError;
  final int lastSuccessfulConnectionTimeMs;
  final int failureCount;
  final int nextReconnectTimeMs;

  ConnectionInformation({
    required this.status,
    required this.isOfflineMode,
    this.lastError,
    this.lastSuccessfulConnectionTimeMs = 0,
    this.failureCount = 0,
    this.nextReconnectTimeMs = 0,
  });

  @override
  String toString() => 'ConnectionInformation(status: $status, '
      'offline: $isOfflineMode, lastError: $lastError, '
      'lastSuccess: $lastSuccessfulConnectionTimeMs, '
      'failures: $failureCount, nextReconnect: $nextReconnectTimeMs)';
}
